subsys/ieee802154/nrf_802154_configurator.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <init.h>
+#include <logging/log.h>
+#include <sys/__assert.h>
+#include <devicetree.h>
+#include <nrf_802154.h>
+#include <nrf_802154_config.h>
+
+LOG_MODULE_REGISTER(nrf_802154_configure);
+
+static uint8_t lna_gain_get(void)
+{
+#if IS_ENABLED(CONFIG_MPSL_FEM_NRF21540_GPIO)
+	return CONFIG_MPSL_FEM_NRF21540_RX_GAIN_DB;
+
+#elif IS_ENABLED(CONFIG_MPSL_FEM_SIMPLE_GPIO)
+#if !DT_NODE_EXISTS(DT_NODELABEL(nrf_radio_fem))
+#error Node with label 'nrf_radio_fem' not found in the devicetree.
+#endif
+	return DT_PROP(DT_NODELABEL(nrf_radio_fem), rx_gain_db);
+
+#else
+	return 0U;
+#endif
+}
+
+static void ccaed_threshold_configure(void)
+{
+	nrf_802154_cca_cfg_t cca_cfg;
+
+	nrf_802154_cca_cfg_get(&cca_cfg);
+
+	/* Overwrite thresholds accounting for RX track gain. */
+	cca_cfg.ed_threshold = NRF_802154_CCA_ED_THRESHOLD_DEFAULT + lna_gain_get();
+	cca_cfg.corr_threshold = NRF_802154_CCA_CORR_THRESHOLD_DEFAULT + lna_gain_get();
+
+	nrf_802154_cca_cfg_set(&cca_cfg);
+}
+
+static int nrf_802154_configure(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	ccaed_threshold_configure();
+
+	return 0;
+}
+
+#if defined(CONFIG_NRF_802154_SER_RADIO)
```

Comment:
```
#define INIT_PRIO (IS_ENABLED(CONFIG_NRF_802154_SER_RADIO)?(CONFIG_NRF_802154_SER_RADIO_INIT_PRIO):(CONFIG_IEEE802154_NRF5_INIT_PRIO))

BUILD_ASSERT(INIT_PRIO < CONFIG_NRF_802154_RADIO_CONFIG_PRIO,
	     "nRF 802.15.4 driver configuration would not be performed after its initialization");
```
will remove duplication

Reply:
btw - I expect it assumes that both things are at the same level (POST_KERNEL), right?

Reply:
Not defined symbols strike again, but now reduction is slightly reduced.
Yes, it is assumed. If this assumption is not satisfied, then broken 802.15.4 configuration would be the least of problems.